pixeltable/env.py

```diff
@@ -816,6 +818,91 @@ def __init_spacy(self) -> None:
         except Exception as exc:
             raise excs.Error(f'Failed to load spaCy model: {spacy_model}') from exc
 
+    def _cleanup(self) -> None:
+        """
+        Internal cleanup method that properly closes all resources and resets state.
+        This is called before destroying the singleton instance.
+        """
+        # Close database connections and sessions
+        if self._current_session is not None:
+            try:
+                self._current_session.close()
+            except Exception as e:
+                _logger.error(f'Error closing session: {e}')
+            self._current_session = None
+
+        if self._current_conn is not None:
+            try:
+                self._current_conn.close()
+            except Exception as e:
+                _logger.error(f'Error closing connection: {e}')
+            self._current_conn = None
+
+        # Stop HTTP server
+        if self._httpd is not None:
+            try:
+                self._httpd.shutdown()
+                self._httpd.server_close()
+            except Exception as e:
+                _logger.error(f'Error stopping HTTP server: {e}')
+            self._httpd = None
+
+        # Stop database server (for local environment)
+        if self._db_server is not None and self._dbms is not None:
+            try:
+                # First terminate all connections to the database
+                if self._db_name is not None:
```

Comment:
Is it possible for `self._db_name` to be `None` even though `self._db_server is not None and self._dbms is not None`? If not, make this into an assertion.

Reply:
self._dbms should always be set if env has initialized. 
_db_name and _db_server will always be set together.
I can add assertions for them

In test setup we just call Env.init_env() even if its not yet set so goal here was to not worry if its initialized correctly or not just reset it. (There were cases due to bugs/issues where env was not initialized correctly for a db but then for another db it was successfully initialized so in this case we want to clear out any previous attempt, rather than restarting lamdba function)